odoo/http.py

```diff
@@ -1479,6 +1483,35 @@ def _dispatch_nodb():
         except werkzeug.exceptions.HTTPException as e:
             return e(environ, start_response)
 
+    def get_dispatch_context_managers(self, request):
+        """ Return the context managers to use for dispatching a request.
+        This includes ``request`` itself, and optionally a profiler.
+        """
+        context_managers = [request]
+        if request.session.profile_session and request.session.db:
+            if 'set_profiling' in request.httprequest.path:
+                _logger.debug("Profiling disabled on route set_profiling")
+            elif request.session.profile_expiration < str(datetime.now()):
+                # avoid having session profiling for too long if user forgets to disable profiling
+                request.session.profile_session = None
+                _logger.warning("Profiling expiration reached, disabling profiling")
+            elif request.httprequest.path.startswith('/longpolling'):  # and odoo.multi_process:
```

Comment:
shouldn't that be based on the process-wide flag `odoo.evented` instead?
```suggestion
            elif odoo.evented:  # profiling doesn't work for greenlets
```

Reply:
Actually you are right, odoo.evented would be less hacky. 

But the final decision was that in any case, profiling longpolling will create a lot of useless entries in ir_profile, evented or not.